site/_posts/2020-05-27-CSI-integration.md

```diff
@@ -0,0 +1,201 @@
+          mountPath: "/mnt/azuredisk"
+  volumes:
+    - name: azuredisk01
+      persistentVolumeClaim:
+        claimName: pvc-azuredisk
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  namespace: csi-app
+  name: pvc-azuredisk
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+  storageClassName: disk.csi.azure.com
+---
+```
+
+`kubectl apply 👆 yaml`
+
+Once our pod is running write some data into `/mnt/azuredisk`, where the CSI volume is mounted.
+
+```bash
+$ kubectl -n csi-app exec -ti csi-nginx bash 
+root@csi-nginx:/# while true; do echo -n "FOOBARBAZ " >> /mnt/azuredisk/foobar; done 
+^C 
+root@csi-nginx:/# cksum /mnt/azuredisk/foobar 
+2279846381 1726530 /mnt/azuredisk/foobar 
+```
+
+## Back Up
+
+Back up the `csi-app` namespace by running the below command
+
+```bash
+$ velero backup create csi-b2 --include-namespaces csi-app --wait 
+Backup request "csi-b2" submitted successfully. 
+Waiting for backup to complete. You may safely press ctrl-c to stop waiting - your backup will continue in the background. 
+.................. 
+Backup completed with status: Completed. You may check for more information using the commands `velero backup describe csi-b2` and `velero backup logs csi-b2`.
+```
+
+## Restore
+
+Before restoring from the backup simulate a disaster by running
+
+```bash
+kubectl delete ns csi-app
+```
+
+Once the namespace has been deleted, restore the `csi-app` from the backup `csi-b2`.
+
+```bash
+$ velero create restore --from-backup csi-b2 --wait 
+Restore request "csi-b2-20200518085136" submitted successfully. 
+Waiting for restore to complete. You may safely press ctrl-c to stop waiting - your restore will continue in the background. 
+.... 
+Restore completed with status: Completed. You may check for more information using the commands `velero restore describe csi-b2-20200518085136` and `velero restore logs csi-b2-20200518085136`. 
+```
+
+Now that the restore has completed and our `csi-nginx` pod is `Running`, confirm that contents of `/mnt/azuredisk/foobar` have been correctly restored.
+
+```bash
+$ kubectl -n csi-app exec -ti csi-nginx bash 
+root@csi-nginx:/# cksum /mnt/azuredisk/foobar 
+2279846381 1726530 /mnt/azuredisk/foobar 
+root@csi-nginx:/# 
+```
+
+## Resources
+
+More details about CSI volume snapshotting and its support in Velero may be found in the following links:
+
+- [Kubernetes 1.17 Feature: Kubernetes Volume Snapshot Moves to Beta](https://kubernetes.io/blog/2019/12/09/kubernetes-1-17-feature-cis-volume-snapshot-beta/) for more information on the CSI beta snapshot APIs.
+- Prerequisites to use this feature is available [on our website](https://velero.io/docs/csi).
+- [Kubernetes CSI docs](https://kubernetes-csi.github.io/docs/sidecar-containers.html): To understand components in a CSI environment
+- [Velero plugin for CSI](https://github.com/vmware-tanzu/velero-plugin-for-csi) for implementation details of the CSI plugin.
```

Comment:
Can we change the link text to say `Velero plugin for CSI snapshots`? This will help make it clear what part of the CSI spec we're supporting.